lib/Wikia/src/DependencyInjection/InjectorInitializer.php

```diff
@@ -12,6 +13,7 @@ public static function init(CacheProvider $cacheProvider = null) {
 			(new InjectorBuilder())
 				->withCache($cacheProvider)
 				->addModule(new PreferenceModule())
+				->addModule(new AttributesModule())
```

Comment:
If you're not familiar with the dependency injection library we just added (which I assume you're not b/c it was just added, I wasn't), this is the entry point.

When MW lurches into existence, this Attributes Module is called which binds a set of interfaces to concrete classes which implement those interfaces. It also binds the class constant `UserAttributes::DEFAULT_ATTRIBUTES` to `\User::getDefaultOptions();`. All this allows the various dependencies to be mocked easily for testing (which you can see in the unit tests below).

Instantiating the client is done through the DI library [here](https://github.com/Wikia/app/pull/7943/files#diff-897fba1dd6b8edc3238762e35d0b7c71R275) which takes care of resolving all the dependencies, and injecting instances of those dependencies.